stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -1418,15 +1418,24 @@ function cond(A::AbstractMatrix, p::Real=2)
     if p == 2
         v = svdvals(A)
         maxv = maximum(v)
-        return maxv == 0.0 ? oftype(real(A[1,1]),Inf) : maxv / minimum(v)
+        return iszero(maxv) ? oftype(real(maxv), Inf) : maxv / minimum(v)
     elseif p == 1 || p == Inf
         checksquare(A)
-        return _cond1Inf(A, p)
+        try
+            Ainv = inv(A)
+            return opnorm(A, p)*opnorm(Ainv, p)
+        catch e
+            if isa(e, LAPACKException) || isa(e, SingularException)
+                TA = real(eltype(A))
+                Tout = typeof((one(TA)*zero(TA) + one(TA)*zero(TA))/one(TA))
+                return convert(Tout, Inf)
```

Comment:
The issue seems to be that `Inf` is not in `Int`, right? We could simplify this a bit to
```julia
TA = float(real(eltype(A)))
return convert(TA, Inf)
```
?

Reply:
Great idea, thanks!